src/adapters/IDB.js

```diff
@@ -81,6 +87,62 @@ export default class IDB extends BaseAdapter {
   }
 
   /**
+   * Batch operations.
+   *
+   * @param  {Function} fn The operation function, which must return a Promise.
+   * @return {Promise}
+   */
+  batch(fn) {
+    return this.open().then(() => {
+      const errors = [], operations = [];
+      const {transaction, store} = this.prepare("readwrite");
+      const batchObject = {
+        get(id) {
+          if (!id) {
+            return Promise.reject(new Error("Id not provided."));
+          }
+          return new Promise((resolve, reject) => {
+            const request = store.get(id);
+            request.onerror = event => reject(new Error(event.target.error));
+            request.onsuccess = () => resolve(request.result);
+          });
+        }
+      };
+      for (let type of Object.keys(BATCH_STORE_METHODS)) {
+        batchObject[type] = data => {
+          const operation = {type, data};
+          operations.push(operation);
+          return operation;
+        };
+      }
+      const result = fn(batchObject);
+      return Promise.resolve(result)
+        .then(result => {
```

Comment:
Actually, `result` is not used in this function, so lines 118-120 can just be `return fn(batchObject).then(() => {`